src/foam/core/reflow/Upload.js

```diff
@@ -95,19 +95,70 @@ foam.CLASS({
 foam.CLASS({
   package: 'foam.core.reflow',
   name: 'Upload',
+  // extends: 'foam.u2.Controller',
 
   requires: [
     'foam.dao.MDAO',
     'foam.lib.csv.CSVParser',
     'foam.core.reflow.ColumnParser',
     'foam.core.reflow.DAOHolder',
     'foam.core.reflow.Mapping',
-    'foam.core.reflow.UploadAgent'
+    'foam.core.reflow.UploadAgent',
+    'foam.core.fs.fileDropZone.FileDropZone',
+    'foam.core.fs.File'
   ],
 
   imports: [ 'currentBlock?', 'eval_?', 'setTimeout' ],
 
+
   properties: [
+    {
+      class: 'FObjectArray',
+      of: 'foam.lang.FObject',
+      name: 'uploadedFiles',
+      factory: function() { return []; },
+      postSet: function(_, n) {
+        if (n && n.length > 0) {
+          this.input = '';
+          this.processUploadedFiles();
+        }
+      },
+      view: function(_, X) {
+        return {
+          class: 'foam.core.fs.fileDropZone.FileDropZone',
+          files$: X.data.uploadedFiles$,
+          supportedFormats: {
+            'text/csv': 'CSV',
+            'application/json': 'JSON',
+            'text/xml': 'XML',
+            'text/plain': 'TXT'
+          },
+          isMultipleFiles: false,
+          title: 'Drag and drop a file here or click to browse',
+          onFilesChanged: X.data.onFilesChanged.bind(X.data)
+        };
+      },
+      visibility: function(input) {
+        return (!input || input.trim() === '') ? 
```

Comment:
```suggestion
        return ( ! input || input.trim() === '' ) ? 
```
style guide ref: https://github.com/kgrgreer/foam3/wiki/CodingStyle
 the spacing should be reviewed throughout this PR

Reply:
done for all of the file